src/Dscp/Educator/Web/Student/Types.hs

```diff
@@ -0,0 +1,95 @@
+
+-- | Datatypes for Student HTTP API
+
+module Dscp.Educator.Web.Student.Types
+       ( Course (..)
+       , Assignment (..)
+       , Submission (..)
+       , Grade (..)
+       , BlkProof (..)
+       , ErrResponse (..)
+       ) where
+
+import Data.Aeson.Options (defaultOptions)
+import Data.Aeson.TH (deriveJSON)
+import Data.Time.Clock (UTCTime)
+import Servant (FromHttpApiData (..))
+
+import Dscp.Core.Address (addrFromText)
+import Dscp.Core.Aeson ()
+import qualified Dscp.Core.Types as Core
+import Dscp.Crypto (Hash, Raw)
+import Dscp.Educator.Txs (PrivateTx (..))
+import Dscp.Educator.Web.Student.Error (APIError)
+import Dscp.Util (fromBase64)
+import Dscp.Util.Aeson (AsByteString, Base64Encoded)
+
+data Course = Course
```

Comment:
Even though these datatypes are in different modules, a different naming would look better IMO. How about `W` prefix? `WCourse`/`WAssignment` etc?

Reply:
Since `purescript-bridge`, I have a subconscious disgust for adapter types with letter prefixes :|
We can do it still, but I'd like to collect more votes for or against.